src/providers/cryptoauthlib/hash.rs

```diff
@@ -8,33 +8,34 @@ use parsec_interface::operations::psa_hash_compute;
 use parsec_interface::requests::{ResponseStatus, Result};
 
 impl Provider {
+    /// Calculate SHA2-256 digest for a given message using CALib.
+    /// Ensure proper return value type.
+    pub fn sha256(&self, msg: &[u8]) -> Result<psa_hash_compute::Result> {
+        let mut hash = vec![0u8; rust_cryptoauthlib::ATCA_SHA2_256_DIGEST_SIZE];
+        match self.device.sha(msg.to_vec(), &mut hash) {
+            rust_cryptoauthlib::AtcaStatus::AtcaSuccess => {
+                Ok(psa_hash_compute::Result { hash: hash.into() })
+            }
+            rust_cryptoauthlib::AtcaStatus::AtcaBadParam => {
+                Err(ResponseStatus::PsaErrorInvalidArgument)
+            }
+            rust_cryptoauthlib::AtcaStatus::AtcaSmallBuffer => {
+                Err(ResponseStatus::PsaErrorBufferTooSmall)
+            }
+            rust_cryptoauthlib::AtcaStatus::AtcaRxNoResponse
+            | rust_cryptoauthlib::AtcaStatus::AtcaRxFail => {
+                Err(ResponseStatus::PsaErrorCommunicationFailure)
+            }
+            _ => Err(ResponseStatus::PsaErrorGenericError),
```

Comment:
I think it would be best to have detailed report from hardware, especially when something goes wrong. I think my correction addresses your comment and leaves enough information for diagnosing any issue that may occur.

Reply:
Ah, sorry, I think I was a bit ambiguous! I meant that these conversions do make sense, but in this case are (were?) attached to a specific function instead of implementing them somewhere where they could be reused.

Reply:
This is how I understood you comment. I checked how these detailed results can be reused and in my opinion their only purpose is to precisely inform what went wrong. Therefore a logged error is even better - it is now obvious that error comes from hardware, not to mention all errors can be logged, not only those imaginable at the time of writing this code.

Reply:
True, I think for the other providers we simply assumed that if such an error occured it was due to a bug in our code, as we try to guarantee that the calls are correct in nature (so, for example, `AtcaBadParam` should never happen). Thanks!